src/components/smart_objects/include/smart_objects/enum_schema_item.h

```diff
@@ -361,60 +403,17 @@ Errors::eType TEnumSchemaItem<EnumType>::validate(
     auto signatures_it = mElementSignatures.find(value);
     if (signatures_it != mElementSignatures.end()) {
       if (!signatures_it->second.empty()) {
-        for (uint i = 0; i < signatures_it->second.size(); i++) {
-          ElementSignature signature = signatures_it->second[i];
-          // Check if signature matches message version
-          if (signature.mSince.is_initialized()) {
-            if (MessageVersion < signature.mSince.get()) {
-              // Msg version predates 'since' field, check next entry
-              continue;
-            } else {
-              if (signature.mUntil.is_initialized() &&
-                  (MessageVersion >= signature.mUntil.get())) {
-                continue;  // Msg version newer than `until` field
-              } else {
-                // Found correct version
-                if (signature.mRemoved) {
-                  // Element was removed for this version
-                  std::string validation_info = "Invalid enum value: " +
-                                                Object.asString() +
-                                                " removed for SyncMsgVersion " +
-                                                MessageVersion.toString();
-                  report__->set_validation_info(validation_info);
-                  return Errors::INVALID_VALUE;
-                }
-                return Errors::OK;  // Mobile msg version falls within specified
-                                    // version range and is not removed
-              }
-            }
-          }  // end if signature.mSince.is_initialized()
-
-          if (signature.mUntil.is_initialized() &&
-              (MessageVersion >= signature.mUntil.get())) {
-            continue;  // Msg version newer than `until` field, check next entry
-          } else {
-            // Found correct version
-            if (signature.mRemoved) {
-              // Element was removed for this version
-              std::string validation_info =
-                  "Enum value : " + Object.asString() +
-                  " removed for SyncMsgVersion " + MessageVersion.toString();
-              report__->set_validation_info(validation_info);
-              return Errors::INVALID_VALUE;
-            }
-            return Errors::OK;  // Mobile msg version falls within specified
-                                // version range and is not removed
-          }
-        }  // End For Loop Version not found.
-        std::string validation_info =
-            "Invalid enum value: " + Object.asString() +
-            " for SyncMsgVersion " + MessageVersion.toString();
-        report__->set_validation_info(validation_info);
-        return Errors::INVALID_VALUE;
+        if (getSignature(signatures_it->second, MessageVersion).mRemoved) {
```

Comment:
should differentiate between cases where the enum was removed and when it just doesn't exist in the spec